src/components/Popover/index.js

```diff
@@ -1,44 +1,77 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import {createPortal} from 'react-dom';
-import {propTypes, defaultProps} from './popoverPropTypes';
+import {withOnyx} from 'react-native-onyx';
+import {propTypes as popoverPropTypes, defaultProps as popoverDefaultProps} from './popoverPropTypes';
 import CONST from '../../CONST';
 import Modal from '../Modal';
 import withWindowDimensions from '../withWindowDimensions';
+import compose from '../../libs/compose';
+import ONYXKEYS from '../../ONYXKEYS';
+
+const propTypes = {
+    isShortcutsModalOpen: PropTypes.bool,
+    ...popoverPropTypes,
+};
+
+const defaultProps = {
+    isShortcutsModalOpen: false,
+    ...popoverDefaultProps,
+};
 
 /*
  * This is a convenience wrapper around the Modal component for a responsive Popover.
  * On small screen widths, it uses BottomDocked modal type, and a Popover type on wide screen widths.
  */
-const Popover = (props) => {
-    if (!props.fullscreen && !props.isSmallScreenWidth) {
-        return createPortal(
+
+class Popover extends React.Component {
```

Comment:
**qq,** any specific reason why we needed this to be moved to a class component?

Wouldn't the following check suffice the requirement?

```
  if ( !this.props.isShortcutsModalOpen) {
      this.props.onClose();
      return null;
  }
```

Reply:
Yeah, the linter was complaining that I should return early 😞

Reply:
>   32:35  error  Prefer an early return to a conditionally-wrapped function body  rulesdir/prefer-early-return

Reply:
I just attempted:

```diff

diff --git a/src/components/Popover/index.js b/src/components/Popover/index.js
index 9a2e20ed3d..8205db0c1e 100644
--- a/src/components/Popover/index.js
+++ b/src/components/Popover/index.js
@@ -1,15 +1,23 @@
 import React from 'react';
+import {withOnyx} from 'react-native-onyx';
 import {createPortal} from 'react-dom';
 import {propTypes, defaultProps} from './popoverPropTypes';
 import CONST from '../../CONST';
 import Modal from '../Modal';
+import compose from '../../libs/compose';
 import withWindowDimensions from '../withWindowDimensions';
+import ONYXKEYS from '../../ONYXKEYS';

 /*
  * This is a convenience wrapper around the Modal component for a responsive Popover.
  * On small screen widths, it uses BottomDocked modal type, and a Popover type on wide screen widths.
  */
 const Popover = (props) => {
+    if (props.isShortcutsModalOpen && props.isVisible) {
+        props.onClose();
+        return null;
+    }
+
     if (!props.fullscreen && !props.isSmallScreenWidth) {
         return createPortal(
             <Modal
@@ -41,4 +49,9 @@ Popover.propTypes = propTypes;
 Popover.defaultProps = defaultProps;
 Popover.displayName = 'Popover';

-export default withWindowDimensions(Popover);
+export default compose(
+    withWindowDimensions,
+    withOnyx({
+        isShortcutsModalOpen: {key: ONYXKEYS.IS_SHORTCUTS_MODAL_OPEN},
+    }),
+)(Popover);

```

and I didn't get any lint errors

<img width="935" alt="image" src="https://user-images.githubusercontent.com/3069065/217907678-938c4e9a-2f1c-47b0-a714-dc78553ac009.png">

Reply:
Ahh you mean why I moved it to a class and not exactly about the if condition?

Reply:
About the class, yeah, I guess we didn't need it anymore, I played with the prevProps at the beginning and that's why I created it for componentDidMount, but then I realized the condition was redundant and worked well without checking any prevProps. I can revert it back now that I am only using the props and nothing else.

Reply:
>  I can revert it back now that I am only using the props and nothing else

yeah, exactly. Thank you.